coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -0,0 +1,150 @@
+      }
+      if (execPlans.size == 1) execPlans.head
+      else StitchRvsExec(qContext, InProcessPlanDispatcher, execPlans.sortWith((x, y) => !x.isInstanceOf[PromQlExec]))
+      // ^^ Stitch RemoteExec plan results with local using InProcessPlanDispatcher
+      // Sort to move RemoteExec in end as it does not have schema
+    }
+  }
+
+
+  def materializeBinaryJoin(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
+
+    val routingKeys = getRoutingKeys(logicalPlan)
+    if (routingKeys.forall(_._2.isEmpty)) localPlanner.materialize(logicalPlan, qContext)
+    else {
+      val offsetMillis = LogicalPlanUtils.getOffsetMillis(logicalPlan)
+      val periodicSeriesTime = getPeriodicSeriesTimeFromLogicalPlan(logicalPlan)
+      val periodicSeriesTimeWithOffset = TimeRange(periodicSeriesTime.startMs - offsetMillis,
+        periodicSeriesTime.endMs - offsetMillis)
+      val lookBackTimeMs = getLookBackMillis(logicalPlan)
+      // Time at which raw data would be retrieved which is used to get partition assignments.
+      // It should have time with offset and lookback as we need raw data at time including offset and lookback.
+      val queryTimeRange = TimeRange(periodicSeriesTimeWithOffset.startMs - lookBackTimeMs,
+        periodicSeriesTimeWithOffset.endMs)
+      // Lhs & Rhs in Binary Join can have different values
+      val routingKeyMap = routingKeys.flatMap(x => x._2.get.map(y => (x._1, y))).toMap
+      val partitions = partitionLocationProvider.getPartitions(routingKeyMap, queryTimeRange).
+        sortBy(_.timeRange.startMs)
+      val partitionName = partitions.head.partitionName
+
+      // Binary Join supported only fro single partition now
+      if (partitions.forall(_.partitionName.equals((partitionName)))) {
+        if (partitionName.equals(localPartition)) localPlanner.materialize(logicalPlan, qContext)
+        else generateRemoteExec(qContext, periodicSeriesTime.startMs / 1000, periodicSeriesTime.endMs / 1000,
+          partitions.head.endPoint)
+      }
+      else throw new UnsupportedOperationException("Binary Join across multiple partitions not supported")
+    }
+  }
+
+  def materializeSeriesKeysFilters(lp: SeriesKeysByFilters, qContext: QueryContext): ExecPlan = {
+   val partitions = partitionLocationProvider.getAuthorizedPartitions(TimeRange(lp.startMs, lp.endMs))
+    val execPlans = partitions.map { p =>
+      if (p.partitionName.equals(localPartition))
+        localPlanner.materialize(lp.copy(startMs = p.timeRange.startMs, endMs = p.timeRange.endMs), qContext)
+      else generateRemoteExec(qContext, p.timeRange.startMs, p.timeRange.endMs, p.endPoint)
+    }
+    if (execPlans.size == 1) execPlans.head
+    else PartKeysDistConcatExec(qContext, InProcessPlanDispatcher,
+      execPlans.sortWith((x, y) => !x.isInstanceOf[PromQlExec]))
+  }
+
+  def materializeLabelValues(lp: LabelValues, qContext: QueryContext): ExecPlan = {
```

Comment:
Similarly avoid duplicate code by moving all the common logic between `materializeSeriesKeysFilters` and `materializeLabelValues` to helper methods